apkid/rules/dex/obfuscators.yara

```diff
@@ -187,3 +187,23 @@ rule allatori_demo : obfuscator
   condition:
     $s and is_dex
 }
+
+rule aamo : obfuscator
+{
+  meta:
+    description = "AAMO"
+    author = "P0r0"
+    url = "https://github.com/necst/aamo"
+    backup_url = "https://github.com/P0r0/aamo"
+    example1 = "c1ef860af0e168f924663630ed3b61920b474d0c8b10e2bde6bfd3769dbd31a8"
+    example2 = "eb0d4e1ba2e880749594eb8739e65aa21b6f7b43798f04b6681065b396c15a78"
+    example3 = "b1e20bf3bdc53972424560e20c6d9ad12e5e47b8ed429a77f4ba5ff6cb92cb27"
+    example4 = "82a570c272579aacdc22410e152f4519738f4e0ececa84e016201c33ad871fa6"
+
+  strings:
+    $a = { 00 0f 63 6f 6e 76 65 72 74 54 6f 53 74 72 69 6e 67 00 } // convertToString
+    $b = { 00 14 67 65 74 53 74 6f 72 61 67 65 45 6e 63 72 79 70 74 69 6f 6e 00 } //getStorageEncryption
```

Comment:
This is part of the device admin API and shouldn't be a rule - https://developer.android.com/reference/android/app/admin/DevicePolicyManager

Reply:
Yes. I naively tried to express what I was reading from the code:
- Condition 1: it always call a function named *convertToString*
- Condition 2: the latter always calls *getStorageEncryption*

Therefore I thought that combining Condition1 and Condition2 would be enough as Condition2 is never used alone.

Reply:
If we could do semantic, dynamic, behavioral rules, then that would be great and we could be a lot more precise. In this instance though, the rule would FP on many app which accessed storage encryption and had any code (perhaps in one of the many libraries) which defined a `convertToString` method.

One thing to try is the bytecode of the decryption routine. This would be the opcodes and some of the constants (and not the registers and not any string, function, etc. indices because those get remapped easily). Feel free to hit me up on slack if you'd like any pointers on this.